crates/solver/src/settlement.rs

```diff
@@ -299,6 +303,78 @@ impl Settlement {
         }
     }
 
+    // Checks whether the settlement prices do not deviate more than max_settlement_price_deviation from the auction prices on certain pairs.
+    pub fn satisfies_price_checks(
+        &self,
+        auction_id: u64,
+        solver_name: &str,
+        external_prices: &ExternalPrices,
+        max_settlement_price_deviation: &Ratio<BigInt>,
+        tokens_to_satisfy_price_test: &Option<HashSet<H160>>,
+    ) -> bool {
+        if matches!(tokens_to_satisfy_price_test, Some(token_list) if token_list.is_empty()) {
+            return true;
+        }
+        // The following check is quadratic in run-time, although a similar check with linear run-time would also be possible.
+        // For the linear implementation, one would have to find a unique scaling factor that scales the external prices into
+        // the settlement prices. Though, this scaling factor is not easy to define, if reference tokens like ETH are missing.
+        // Since the checks would heavily depend on this scaling factor, and its
+        // derivation is non-trivial, we decided to go for the implementation with quadratic run time. Settlements
+        // will not have enough tokens, such that the run-time is important.
+        !self
+            .clearing_prices()
+            .iter()
+            .combinations(2)
+            .any(|clearing_price_vector_combination| {
+                let sell_token_info = clearing_price_vector_combination.get(0).unwrap();
+                let sell_token = sell_token_info.0;
+                let clearing_price_sell_token = sell_token_info.1.to_big_rational();
+                let buy_token_info = clearing_price_vector_combination.get(1).unwrap();
+                let buy_token = buy_token_info.0;
+                let clearing_price_buy_token = buy_token_info.1.to_big_rational();
+
+                if matches!(tokens_to_satisfy_price_test, Some(token_list) if (!token_list.contains(sell_token)) || !token_list.contains(buy_token))
+                {
+                    return false;
```

Comment:
Shouldn't this return true? Otherwise we filter out all settlements that trade a token that is not in the list.

Reply:
You could remove the outer negation, change `any` to `all`, negate inside the clojure. Double check this but I think it is equivalent.

Reply:
But any is more performant, isn't it? Then it can stop early.

Reply:
`all()` can also return early if it finds an item which does NOT satisfy the predicate.

Reply:
You guys are right. Thanks!